lib/UI/homepage.dart

```diff
@@ -0,0 +1,255 @@
+          SliverPersistentHeader(
+            delegate: MySliverAppBar(
+              onEyePressed: () {
+                setState(() {
+                  _showCompleted = !_showCompleted;
+                });
+              },
+              completedCount: _tasks.where((task) => task['completed']).length,
+            ),
+            pinned: true,
+          ),
+          SliverList(
+            delegate: SliverChildBuilderDelegate(
+              (BuildContext context, int index) {
+                if (index >= _tasks.length) {
+                  return Container(
+                    margin: const EdgeInsets.only(
+                      left: 8,
+                      right: 8,
+                      bottom: 36,
+                    ),
+                    decoration: BoxDecoration(
+                      color: Colors.white,
+                      boxShadow: [
+                        BoxShadow(
+                          color: Colors.grey.withOpacity(0.3),
+                          spreadRadius: 1,
+                          blurRadius: 1,
+                          offset: const Offset(0, 2),
+                        )
+                      ],
+                      borderRadius: const BorderRadius.only(
+                        bottomLeft: Radius.circular(10),
+                        bottomRight: Radius.circular(10),
+                      ),
+                    ),
+                    child: ListTile(
+                      onTap: () {
+                        Navigator.push(
+                          context,
+                          MaterialPageRoute(
+                            builder: (context) =>
+                                const TaskPage(created: false, task: {}, index: -1),
+                          ),
+                        );
+                      },
+                      title: const Text(
+                        "Новая задача...",
+                        style: TextStyle(
+                          fontSize: 16,
+                          height: 20 / 16,
+                          color: Color.fromRGBO(0, 0, 0, 0.3),
+                        ),
+                      ),
+                      leading: const Icon(
+                        Icons.add,
+                        color: Colors.white,
+                      ),
+                    ),
+                  );
+                }
+
+                final task = _tasks[index];
+                if (!_showCompleted && task['completed']) {
+                  return Container();
+                }
+                return TaskItem(
+                  task: task,
+                  onDismissed: (direction) {
+                    if (direction == DismissDirection.endToStart) {
+                      setState(() {
+                        _tasks.removeAt(index);
+                        MyLogger.d("Удаление");
+                      });
+                    } else {
+                      setState(() {
+                        task['completed'] = true;
```

Comment:
Чтобы по свайпу вправо таска не только отмечалась сделанной, но и могла превращаться из "не сделанной" в "сделанную", можно немного отредактировать код:
```suggestion
                        task['completed'] = !task['completed'];
```